Gems/Atom/RHI/Code/Include/Atom/RHI.Reflect/MemoryUsage.h

```diff
@@ -40,7 +40,7 @@ namespace AZ
             //! This helper function checks whether a new allocation is within the budget
             bool CanAllocate(size_t sizeInBytes) const
             {
-                if (m_budgetInBytes && (m_totalResidentInBytes + sizeInBytes) > m_budgetInBytes)
+                if (m_budgetInBytes && (m_usedResidentInBytes + sizeInBytes) > m_budgetInBytes)
```

Comment:
It's hard to release physical memory immediately since it depends on the heap allocator. And if we are moving to VMA and d3d12MA, it's possible we won't be able to keep an accurate heap memory usage per pool. So I think using moving to use  used memory to track memory usage is more realistic and simpler. 
We are still maintain the value of m_totalResidentInBytes before moving to d3d12MA or VMA. I am not sure about afterwards.